test/aaa_profiling/test_threading.py

```diff
@@ -0,0 +1,228 @@
+import threading
+
+import sqlalchemy as sa
+from sqlalchemy import create_engine
+from sqlalchemy import Integer
+from sqlalchemy import String
+from sqlalchemy import testing
+from sqlalchemy.orm import scoped_session
+from sqlalchemy.orm import sessionmaker
+from sqlalchemy.testing import eq_
+from sqlalchemy.testing import fixtures
+from sqlalchemy.testing import in_
+from sqlalchemy.testing.schema import Column
+from sqlalchemy.testing.schema import Table
+
+NUM_THREADS = 10
+
+
+def run_threaded(func, *thread_args, nthreads=NUM_THREADS, **thread_kwargs):
+    barrier = threading.Barrier(nthreads)
+    results = []
+    errors = []
+
+    def thread_func(*args, **kwargs):
+        name = threading.current_thread().name
+        barrier.wait()
```

Comment:
ive found an existing test that legitimately fails under free-threaded builds.   So fixed the error, and am now porting the test to this new fixture.   However, I observe that the failures for the tests dont occur if this "barrier" logic is present.   so I'm wondering if maybe each test should be run with and without the barrier.

Reply:
> to fix, I've had to put a mutex around that method (which I've made conditional on if free threading is enabled).

Yeah, this looks like a race on the gil-enabled build as well though.